src/background/service/autoLock.ts

```diff
@@ -1,44 +1,48 @@
 import { isManifestV3 } from '@/utils/env';
 import preference from './preference';
 import browser from 'webextension-polyfill';
-import { ALARMS_AUTO_LOCK } from '../utils/alarms';
+
+const AUTO_LOCK_AT_KEY = 'autoLockAt';
 
 class AutoLockService {
   timer: ReturnType<typeof setTimeout> | null = null;
+  autoLockAt: number | null = null;
 
   onAutoLock?: () => void;
 
   constructor({ onAutoLock }: { onAutoLock?: () => void } = {}) {
     this.onAutoLock = onAutoLock;
   }
 
-  resetTimer() {
+  async syncAutoLockAt() {
+    if (!isManifestV3) return;
+    const value = await browser.storage.session.get(AUTO_LOCK_AT_KEY);
+    const autoLockAt = value[AUTO_LOCK_AT_KEY];
+    if (autoLockAt) {
+      if (Date.now() >= autoLockAt) {
+        this.onAutoLock?.();
+      } else {
+        this.autoLockAt = autoLockAt;
```

Comment:
need set timer
`this.timer = setTimeout(() => this.onAutoLock?.(), autoLockAt - Date.now())`